src/Countries/Wales.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace Spatie\Holidays\Countries;
+
+use Carbon\CarbonImmutable;
+
+class Wales extends Country
+{
+    public function countryCode(): string
+    {
+        return 'uk-cym';
+    }
+
+    protected function christmasDay(int $year): array
+    {
+        $christmasDay = new CarbonImmutable($year . "-12-25", 'Europe/London');
+        $key = 'Christmas Day';
+
+        if ($christmasDay->isSaturday()) {
+            $key .= ' (substitute day)';
+            $christmasDay = $christmasDay->next('monday');
+        }
+
+        if ($christmasDay->isSunday()) {
+            $key .= ' (substitute day)';
+            $christmasDay = $christmasDay->next('tuesday');
+        }
+
+        return [$key => $christmasDay];
+    }
```

Comment:
Great thought for this scenario.

Is it possible you move chrismas, boxing and new year to a generic file so other countries can inherit it.

Reply:
I'm not sure about this as different countries might have different rules when either fall on a weekend.